### PR TITLE
Glossary

### PR DESCRIPTION
**Pull request by elkevorh**
Glossary - added terms and definitions

### CODE REVIEW

File: src/concepts/terminology.md [hunk 1]
```diff
@@ -1,12 +1,236 @@
...
-* **Derived Dataset**: A processed dataset, such as `main_summary` or the
-  `clients_daily` dataset
-* **Session**: The time from when a Firefox browser starts until it shuts down
-* **Subsession**: `Sessions` are split into `subsessions` when a 24 hour threshold is crossed or an environment change occurs
-([more details](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/concepts/sessions.html?highlight=subsession))
+## Table of Contents
+
+This glossary provides definitions for commonly-used terms at Mozilla.
+
+<!-- TOC -->
```
Comment (wlach) [inline src/concepts/terminology.md:7]:
```suggestion
<!-- toc -->
```

File: src/concepts/terminology.md [hunk 2]
```diff
@@ -1,12 +1,236 @@
...
+        - [-Week Retention](#-week-retention)
+    - [schema](#schema)
+    - [session](#session)
+    - [structured ingestion](#structured-ingestion)
+    - [scheduled query](#scheduled-query)
+    - [sql.telemetry.mozilla.org](#sqltelemetrymozillaorg)
+    - [STMO](#stmo)
+    - [Telemetry](#telemetry)
+        - [WAU](#wau)
+
```
Comment (wlach) [inline src/concepts/terminology.md:51]: This should be automatically generated by the table of contents:

```suggestion
```

File: src/concepts/terminology.md [hunk 3]
```diff
@@ -1,12 +1,236 @@
...
+<!-- /TOC -->
```
Comment (wlach) [inline src/concepts/terminology.md:52]:
```suggestion
```

File: src/concepts/terminology.md [hunk 4]
```diff
@@ -1,12 +1,236 @@
...
+
+A third-party product used by several teams within Mozilla for analysis of user events.
+
+## BigQuery
+
+[BigQuery](https://cloud.google.com/bigquery) is Google Cloud's managed data warehouse. Most of the data described on this site is stored and queried using BigQuery. See the [Accessing and working with BigQuery](https://docs.telemetry.mozilla.org/cookbooks/bigquery.html0) article for more details.
+
+## Client ID
+
+The id property of the Firefox browser returns the universally unique identifier of the Client object.
```
Comment (wlach) [inline src/concepts/terminology.md:72]:
```suggestion
A unique id identifying the client who sent a [ping](#ping).
```

File: src/concepts/terminology.md [hunk 5]
```diff
@@ -1,12 +1,236 @@
...
+
+## Data Analyst
+
+See [Analyst](#Analyst)
+
```
Comment (wlach) [inline src/concepts/terminology.md:77]:
```suggestion
```

File: src/concepts/terminology.md [hunk 6]
```diff
@@ -1,12 +1,236 @@
...
+
+## Data Practioner
+
+See [Analyst](#Analyst)
+
+## Data Scientist
+
+Data scientists at Mozilla are a team with a broad array of technical backgrounds and a core set of common professional skills: 1) applying statistical methods to noisy data to answer questions about what, how, or why something is happening; 2) transform unstructured data into usable metrics and models; and 3) augmenting strategic product and decision-making with empirical evidence created and curated by the team.
```
Comment (wlach) [inline src/concepts/terminology.md:85]: We should probably double check this definition with data science, but meanwhile we can improve the formatting a bit:

```suggestion
Data scientists at Mozilla are a team with a broad array of technical backgrounds and a core set of common professional skills: 

* applying statistical methods to noisy data to answer questions about what, how, or why something is happening
* transform unstructured data into usable metrics and models
* augmenting strategic product and decision-making with empirical evidence created and curated by the team.
```

File: src/concepts/terminology.md [hunk 7]
```diff
@@ -1,12 +1,236 @@
...
+
+## dataset
```
Comment (wlach) [inline src/concepts/terminology.md:87]:
```suggestion
## Dataset
```

File: src/concepts/terminology.md [hunk 8]
```diff
@@ -1,12 +1,236 @@
...
+
+A set of data, which includes ping data, derived datasets, etc.; sometimes it is used synonymously with “table”; sometimes it is used technically to refer to a [BigQuery](https://cloud.google.com/bigquery/docs/datasets-intro) dataset, which represents a container for one or more tables.
```
Comment (wlach) [inline src/concepts/terminology.md:89]: We should probably add a definition for table as well, now that I look at it.

```suggestion
A set of data, which includes ping data, derived datasets, etc.; sometimes it is used synonymously with “table”; sometimes it is used technically to refer to a [BigQuery](#bigquery) dataset, which represents a container for one or more tables.
```

File: src/concepts/terminology.md [hunk 9]
```diff
@@ -1,12 +1,236 @@
...
+
+### derived dataset
+
+A processed dataset, such as `main_summary` or the `clients_daily` dataset
+
+
```
Comment (wlach) [inline src/concepts/terminology.md:95]: Seems to be a less refined version of the definition below, guessing this got included by mistake.

```suggestion
```

File: src/concepts/terminology.md [hunk 10]
```diff
@@ -1,12 +1,236 @@
...
+## DAU
+
+The number of unique profiles active on each day.
+
+## derived dataset
```
Comment (wlach) [inline src/concepts/terminology.md:100]:
```suggestion
## Derived Dataset
```

File: src/concepts/terminology.md [hunk 11]
```diff
@@ -1,12 +1,236 @@
...
+
+A processed dataset, such as the clients_daily dataset. This is in contrast to a raw dataset, which is typically a collection of pings. This data can be computed from some lower-level data, but is stored separately to save the cost and time of re-computing it.
```
Comment (wlach) [inline src/concepts/terminology.md:102]:
```suggestion
A processed dataset, such as [Clients Daily](..datasets/batch_view/clients_daily/reference.md). At Mozilla, this is in contrast to a raw ping table which represents (more or less) the raw data submitted by our users.
```

File: src/concepts/terminology.md [hunk 12]
```diff
@@ -1,12 +1,236 @@
...
+
+## [Fenix](https://mana.mozilla.org/wiki/display/FIREFOX/Fenix+Home+Page)
+
+[Firefox](https://play.google.com/store/apps/details?id=org.mozilla.firefox&hl=en_US) for Android browser.
+
+## FxA
+
+[Firefox Accounts](https://www.mozilla.org/en-CA/firefox/accounts/)
+
```
Comment (wlach) [inline src/concepts/terminology.md:111]: Let's skip these non-data topics. We can refer people to the wiki for more information on them.

```suggestion
```

File: src/concepts/terminology.md [hunk 13]
```diff
@@ -1,12 +1,236 @@
...
+## Glean
+
+Glean is Mozilla’s product analytics & telemetry solution that provides a consistent experience and behavior across all of our products. Most of Mozilla's mobile apps, including Fenix, have been adapted to use the Glean SDK. For more information, see the [Glean Overview](https://docs.telemetry.mozilla.org/concepts/glean/glean.html).
```
Comment (wlach) [inline src/concepts/terminology.md:114]:
```suggestion
Glean is Mozilla’s product analytics & telemetry solution that provides a consistent experience and behavior across all of our products. Most of Mozilla's mobile apps, including Fenix, have been adapted to use the Glean SDK. For more information, see the [Glean Overview](./glean/glean.html).
```

File: src/concepts/terminology.md [hunk 14]
```diff
@@ -1,12 +1,236 @@
...
+
+## GCP
+
+Google Cloud Platform (GCP) is a suite of cloud-computing services that runs on the same infrastructure that Google uses internally for its end-user products.
+
+## GeoIP
+
+IP Geolocation involves attempting to discover the location of an IP address in the real world. IP addresses are assigned to an organization, and as these are ever-changing associations, it can be difficult to determine exactly where in the world an IP address is located. Mozilla’s ingestion infrastructure attempts to perform GeoIP lookup during the data decoding process and subsequently discards the IP address before the message arrives in long-term storage.
+
+## ingestion
```
Comment (wlach) [inline src/concepts/terminology.md:124]:
```suggestion
## Ingestion
```

File: src/concepts/terminology.md [hunk 15]
```diff
@@ -1,12 +1,236 @@
...
+
+Mozilla's core data platform has been built to support structured ingestion of arbitrary JSON payloads whether they come from browser products on client devices or from server-side applications that have nothing to do with Firefox; any team at Mozilla can hook into structured ingestion by defining a schema and registering it with pipeline. Once a schema is registered, everything else is automatically provisioned, from an HTTPS endpoint for accepting payloads to a set of tables in BigQuery for holding the processed data.
+
+## Intensity
+
+Intuitively, how many days per week do users use the product? Among profilesactive at least once in the week ending on the date specified, the number of days on average they were active during that one-week window.
+
+## metric
```
Comment (wlach) [inline src/concepts/terminology.md:132]:
```suggestion
## Metric
```

File: src/concepts/terminology.md [hunk 16]
```diff
@@ -1,12 +1,236 @@
...
+
+## MAU
+
+Monthly Active Users - the number of unique profiles active in the 28-day period ending on a given day. The number of unique profiles active at least once during the 28-day window
+ending on the specified day.
+
+## Notebook Interface
+
+JupyterLab is a web-based interactive development environment for Jupyter notebooks, code, and data. It configures and arranges the user interface to support a wide range of workflows in data science, scientific computing, and machine learning. JupyterLab is extensible and modular and can write plugins that add new components and integrate with existing ones.
+
```
Comment (wlach) [inline src/concepts/terminology.md:144]: Let's skip this for now.

```suggestion
```

File: src/concepts/terminology.md [hunk 17]
```diff
@@ -1,12 +1,236 @@
...
+## ping
```
Comment (wlach) [inline src/concepts/terminology.md:145]:
```suggestion
## Ping
```

File: src/concepts/terminology.md [hunk 18]
```diff
@@ -1,12 +1,236 @@
...
+
+A ping represents a message that is sent from the Firefox browser to Mozilla’s Telemetry servers. It typically includes information about the browser’s state, user actions, etc.
+[for more details](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/data/common-ping.html)
+
+
+## ping table
```
Comment (wlach) [inline src/concepts/terminology.md:151]:
```suggestion
## Ping Table
```

File: src/concepts/terminology.md [hunk 19]
```diff
@@ -1,12 +1,236 @@
...
+
+A set of pings that is stored in BigQuery as a table.
```
Comment (wlach) [inline src/concepts/terminology.md:153]:
```suggestion
A set of pings that is stored in a BigQuery table. See article on [raw ping datasets](../datasets/pings.md).
```

File: src/concepts/terminology.md [hunk 20]
```diff
@@ -1,12 +1,236 @@
...
+
+## pipeline
```
Comment (wlach) [inline src/concepts/terminology.md:155]:
```suggestion
## Pipeline
```

File: src/concepts/terminology.md [hunk 21]
```diff
@@ -1,12 +1,236 @@
...
+
+Mozilla’s data pipeline, which is used to collect Telemetry data from Mozilla’s products and logs from various services.
+The bulk of the data that is handled by this pipeline is Firefox Telemetry data. The same tool-chain is used to collect, store, and analyze data that comes from many sources.
+
+The code for the ingestion pipeline is available in the [gcp-ingestion](https://github.com/mozilla/gcp-ingestion) repository.
+
```
Comment (wlach) [inline src/concepts/terminology.md:161]: Let's skip the code  reference

```suggestion
```

File: src/concepts/terminology.md [hunk 22]
```diff
@@ -1,12 +1,236 @@
...
+
+## ping table
+
+A set of pings that is stored in BigQuery as a table.
+
+## pipeline
+
+Mozilla’s data pipeline, which is used to collect Telemetry data from Mozilla’s products and logs from various services.
+The bulk of the data that is handled by this pipeline is Firefox Telemetry data. The same tool-chain is used to collect, store, and analyze data that comes from many sources.
```
Comment (wlach) [inline src/concepts/terminology.md:158]:
```suggestion
The bulk of the data that is handled by this pipeline is Firefox Telemetry data. The same tool-chain is used to collect, store, and analyze data that comes from many sources.
For more information, see [An overview of Mozilla’s Data Pipeline](./pipeline/gcp_data_pipeline.html).
```

File: src/concepts/terminology.md [hunk 23]
```diff
@@ -1,12 +1,236 @@
...
+
+The code for the ingestion pipeline is available in the [gcp-ingestion](https://github.com/mozilla/gcp-ingestion) repository.
+
+## probe
```
Comment (wlach) [inline src/concepts/terminology.md:162]:
```suggestion
## Probe
```

File: src/concepts/terminology.md [hunk 24]
```diff
@@ -1,12 +1,236 @@
...
+
+Measurements for a specific aspect of Firefox are called probes. A single telemetry ping sends many different probes. Probes are either Histograms (recording distributions of data points) or Scalars (recording a single value).
+
+You can search for details about probes by using the [Probe Dictionary](https://probes.telemetry.mozilla.org/). For each probe, the probe dictionary provides:
+
+* A description of the probe
+* When a probe started being collected
+* Whether data from this probe is collected in the release channel
+
+## profile
```
Comment (wlach) [inline src/concepts/terminology.md:172]:
```suggestion
## Profile
```

File: src/concepts/terminology.md [hunk 25]
```diff
@@ -1,12 +1,236 @@
...
+
+All of the changes a user makes in Firefox, like the home page, what toolbars you use, installed addons, saved passwords and your bookmarks, are all stored in a special folder, called a profile. Telemetry stores archived and pending pings in the profile directory as well as metadata like the client ID. See also [Profile Creation](https://docs.telemetry.mozilla.org/concepts/profile/profile_creation.html)
```
Comment (wlach) [inline src/concepts/terminology.md:174]:
```suggestion
All of the changes a user makes in Firefox, like the home page, what toolbars you use, installed addons, saved passwords and your bookmarks, are all stored in a special folder, called a profile. Telemetry stores archived and pending pings in the profile directory as well as metadata like the [client id](#client-id). See also [Profile Creation](./profile/profile_creation.html)
```

File: src/concepts/terminology.md [hunk 26]
```diff
@@ -1,12 +1,236 @@
...
+
+## query
+
+TBD
+
+## retention
```
Comment (wlach) [inline src/concepts/terminology.md:180]:
```suggestion
## Retention
```

File: src/concepts/terminology.md [hunk 27]
```diff
@@ -1,12 +1,236 @@
...
+
+* A description of the probe
+* When a probe started being collected
+* Whether data from this probe is collected in the release channel
+
+## profile
+
+All of the changes a user makes in Firefox, like the home page, what toolbars you use, installed addons, saved passwords and your bookmarks, are all stored in a special folder, called a profile. Telemetry stores archived and pending pings in the profile directory as well as metadata like the client ID. See also [Profile Creation](https://docs.telemetry.mozilla.org/concepts/profile/profile_creation.html)
+
+## query
```
Comment (wlach) [inline src/concepts/terminology.md:176]:
```suggestion
## Query
```

File: src/concepts/terminology.md [hunk 28]
```diff
@@ -1,12 +1,236 @@
...
+
+TBD
```
Comment (wlach) [inline src/concepts/terminology.md:178]: Let's take a stab at defining this.

```suggestion
Typically refers to a query written in the SQL syntax, run on (for example) [STMO](#stmo).
```

File: src/concepts/terminology.md [hunk 29]
```diff
@@ -1,12 +1,236 @@
...
+### 1-Week New Profile Retention
+
+Among new profiles created on the day specified, what proportion (out of 1) are
+active during the week beginning one week after the day specified.
+
+### 1-Week Retention
+
+Among profiles that were active at least once in the week starting on the
+specified day, what proportion (out of 1) are active during the following week.
+
```
Comment (wlach) [inline src/concepts/terminology.md:197]: Either duplicates or overly specific for a terminology section.

```suggestion
```

File: src/concepts/terminology.md [hunk 30]
```diff
@@ -1,12 +1,236 @@
...
+## schema
+
+A schema is the organization or structure for a database. The activity of data modeling leads to a schema. Mozilla’s source schema is available on the [mozilla-services/socorro GitHub repository](https://raw.githubusercontent.com/mozilla-services/socorro/main/socorro/schemas/crash_report.json).
```
Comment (wlach) [inline src/concepts/terminology.md:200]: I think this isn't quite right. Here's another stab at the concept

```suggestion
A schema is the organization or structure for our data. We use schemas at many levels (in data ingestion and storage) to make sure the data we submit is valid and possible to be processed efficiently.
```

File: src/concepts/terminology.md [hunk 31]
```diff
@@ -1,12 +1,236 @@
...
+
+## session
```
Comment (wlach) [inline src/concepts/terminology.md:202]:
```suggestion
## Session
```

File: src/concepts/terminology.md [hunk 32]
```diff
@@ -1,12 +1,236 @@
...
+
+The period of time that it takes between Firefox being started until it is shut down.
```
Comment (wlach) [inline src/concepts/terminology.md:204]:
```suggestion
The period of time that it takes between Firefox being started until it is shut down. See also [subsession](#subsession).
```

File: src/concepts/terminology.md [hunk 33]
```diff
@@ -1,12 +1,236 @@
...
+
+* **Subsession**: `Sessions` are split into `subsessions` after every 24-hour time period has passed or the environment has changed.
```
Comment (wlach) [inline src/concepts/terminology.md:206]:
```suggestion
## Subsession

In Firefox, [Sessions](#sessions) are split into subsessions after every 24-hour time period has passed or the environment has changed.
```

File: src/concepts/terminology.md [hunk 34]
```diff
@@ -1,12 +1,236 @@
...
+([for more details](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/concepts/sessions.html?highlight=subsession))
+
+## structured ingestion
+
+Mozilla’s term for the URI scheme that is used for submitting data to the Telemetry pipeline; contrast this term with “telemetry ingestion” which represents an older URI scheme that is still supported for compatibility with desktop Firefox telemetry
+
```
Comment (wlach) [inline src/concepts/terminology.md:212]: Let's skip this one, it's hard to explain and not necessary for most people to know.

```suggestion
```

File: src/concepts/terminology.md [hunk 35]
```diff
@@ -1,12 +1,236 @@
...
+## scheduled query
+
+At Mozilla, queries are written in /sql. The directory structure is based on the destination table: /sql/{dataset_id}/{table_name}
+You can schedule queries in [bigquery-etl](https://github.com/mozilla/bigquery-etl) in [Airflow](https://github.com/mozilla/telemetry-airflow) that run regularly with the results written to a table.
+
+* [In bigquery-etl](https://docs.telemetry.mozilla.org/cookbooks/bigquery-airflow.html#in-bigquery-etl)
+* [Other considerations](https://docs.telemetry.mozilla.org/cookbooks/bigquery-airflow.html#other-considerations)
+
```
Comment (wlach) [inline src/concepts/terminology.md:220]: Let's skip this for now.

```suggestion
```

File: src/concepts/terminology.md [hunk 36]
```diff
@@ -1,12 +1,236 @@
...
+## sql.telemetry.mozilla.org
```
Comment (wlach) [inline src/concepts/terminology.md:221]:
```suggestion
## STMO (sql.telemetry.mozilla.org)
```

File: src/concepts/terminology.md [hunk 37]
```diff
@@ -1,12 +1,236 @@
...
+
+A service for creating queries and dashboards. See [STMO under analysis tools](https://docs.telemetry.mozilla.org/tools/interfaces.html#sqltelemetrymozillaorg-stmo).
```
Comment (wlach) [inline src/concepts/terminology.md:223]:
```suggestion
A service for creating queries and dashboards. See [STMO under analysis tools](../tools/interfaces.html#sqltelemetrymozillaorg-stmo).
```

File: src/concepts/terminology.md [hunk 38]
```diff
@@ -1,12 +1,236 @@
...
+
+## STMO
+
+As you use Firefox, Telemetry measures and collects non-personal information, such as performance, hardware, usage and customizations. It then sends this information to Mozilla on a daily basis and we use it to improve Firefox. 
+
```
Comment (wlach) [inline src/concepts/terminology.md:228]:
```suggestion
```

**Review by wlach — CHANGES_REQUESTED** [review]
Hi Elke, a bunch of suggestions but mostly looks good! Tell me what you think and I can merge it next week.

**Comment by willkg** [comment]
@wlach ^^^ That rebases against master tip and addresses all your comments.

**Comment by willkg** [comment]
@wlach: Some of these are spelling errors, but others aren't. How do you deal with the non-errors?

```
$ scripts/spell_check.sh
    src/concepts/terminology.md
       10 | ## AET 
       12 | Ecosystem Telemetry; see the [PRD](https://docs.google.com/docu 
       34 | ## Data Practioner 
       68 | IP Geolocation involves attempting to discov 
       76 |  users use the product? Among profilesactive at least once in the week end 
>> 5 spelling errors found in 155 files
```

**Comment by wlach** [comment]
> @wlach: Some of these are spelling errors, but others aren't. How do you deal with the non-errors?

Sorry for the late response! GitHub needs needinfo, apparently.

You can add new spellings to `.spelling`. See: https://github.com/mozilla/data-docs#spell-checking

**Comment by willkg** [comment]
@wlach Thank you for that! This is ready for another review.

**Review by wlach — APPROVED** [review]
This looks good, there are a few things we should probably also do (e.g. link out to https://wiki.mozilla.org/Acronyms) but that can be done in another PR

**Pull request merged**